v2/samples/azureiotedge-simulated-temperature-sensor/Program.cs

```diff
@@ -0,0 +1,198 @@
+            store.Close();
+        }
+
+
+        /// <summary>
+        /// Initializes the DeviceClient and sets up the callback to receive
+        /// messages containing temperature information
+        /// </summary>
+        static async Task Init(string connectionString, bool bypassCertVerification = false)
+        {
+            Console.WriteLine("Connection String {0}", connectionString);
+
+            var mqttSetting = new MqttTransportSettings(TransportType.Mqtt_Tcp_Only);
+            // During dev you might want to bypass the cert verification. It is highly recommended to verify certs systematically in production
+            if (bypassCertVerification)
+            {
+                mqttSetting.RemoteCertificateValidationCallback = (sender, certificate, chain, sslPolicyErrors) => true;
+            }
+            ITransportSettings[] settings = { mqttSetting };
+
+            // Open a connection to the Edge runtime
+            var ioTHubModuleClient = DeviceClient.CreateFromConnectionString(connectionString, settings);
+            await ioTHubModuleClient.OpenAsync();
+            Console.WriteLine("IoT Hub module client initialized.");
+
+            var moduleTwin = await ioTHubModuleClient.GetTwinAsync();
+            var moduleTwinCollection = moduleTwin.Properties.Desired;
+            desiredProperties = new DesiredProperties();
+            desiredProperties.UpdateDesiredProperties(moduleTwinCollection);
+
+            // callback for updating desired properties through the portal or rest api
+            await ioTHubModuleClient.SetDesiredPropertyUpdateCallbackAsync(onDesiredPropertiesUpdate, null);
+
+            // this direct method will allow to reset the temperature sensor values back to their initial state
+            await ioTHubModuleClient.SetMethodHandlerAsync("reset", ResetMethod, null);
+
+            // we don't pass ioTHubModuleClient as we're not sending any messages out to the message bus
+            await ioTHubModuleClient.SetInputMessageHandlerAsync("control", ControlMessageHandler, null);
+
+            while(true)
+            {
+                try
+                {
+                    if(desiredProperties.SendData)
+                    {
+                        counter++;
+                        if(counter == 1)
+                        {
+                            // first time execution needs to reset the data factory
+                            IsReset = true;
+                        }
+
+                        var messageBody = TemperatureDataFactory.CreateTemperatureData(counter, generationPolicy, IsReset);
+                        IsReset = false;
+                        var messageString = JsonConvert.SerializeObject(messageBody);
+                        var messageBytes = Encoding.UTF8.GetBytes(messageString);
+                        var message = new Message(messageBytes);
+
+                        await ioTHubModuleClient.SendEventAsync("temperatureOutput", message);
```

Comment:
We should move the SendData part to a separate method called from Main. Otherwise Init will never return.